python3Packages.gcsfs: init at 2021.04.0

nbren12 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Add gcsfs, a useful tool for working with Google Cloud Storage. gcsfs implements the interfaces described in fsspec, which is in nixpkgs. I needed to bump the versions fsspec to work with the latest release of gcsfs. These packages are developed in unison, so usually need very similar versions.

I sometimes contribute to this project and am in contact with its primary author.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

legendofmiracles commented:
Make another commit for adding the maintainer `maintainers: add nbren12`

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -6977,6 +6977,12 @@
     githubId = 818502;
     name = "Nathan Yong";
   };
+  nbren12 = {
```
Please cherry-pick the commit into the PR where you are adding yourself as a maintainer of a package.

nbren12 replied:
I split this change into a separate commit: e0b51cc7ae4e59a389781ab373c9acd1d78ae017.


> Please cherry-pick the commit into the PR

Confused about the wording here. Do you mean "a new PR" or "the PR"? For now, I will assume that "The PR" refers to this one.

SuperSandro2000 replied:
I think I picked the wrong default answer.

The correct one is:

Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

SuperSandro2000 commented on pkgs/development/python-modules/fsspec/default.nix:
```diff
@@ -1,28 +1,23 @@
-, numpy
-, stdenv
-}:
+{ lib, buildPythonPackage, fetchFromGitHub, pythonOlder, pytestCheckHook, numpy
+, stdenv, aiohttp, pytest-vcr, requests }:
```
Please revert this change to keep the diff smaller.

nbren12 replied:
done.

SuperSandro2000 reviewed (changes requested):
Please include the version number in the init commit.

dotlambda commented on pkgs/development/python-modules/fsspec/default.nix:
```diff
@@ -33,6 +28,9 @@ buildPythonPackage rec {
     # resolved in https://github.com/intake/filesystem_spec/issues/432 and
     # can be enabled again from version 0.8.4
     "test_pathobject"
```
Can this be removed?

nbren12 replied:
I removed it. The issue has indeed been resolved on the fsspec side.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 120173` at 023c823f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.clifford</li>
    <li>python38Packages.dask-jobqueue</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.rl-coach</li>
    <li>python39Packages.dask-jobqueue</li>
    <li>python39Packages.dask-xgboost</li>
    <li>python39Packages.rl-coach</li>
  </ul>
</details>
<details>
  <summary>16 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.caffe</li>
    <li>python38Packages.dask-glm</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.imgaug</li>
    <li>python38Packages.intake</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.caffe</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>54 packages built successfully:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.batchgenerators</li>
    <li>python38Packages.dask</li>
    <li>python38Packages.dask-gateway</li>
    <li>python38Packages.dask-image</li>
    <li>python38Packages.dask-mpi</li>
    <li>python38Packages.datasets</li>
    <li>python38Packages.distributed</li>
    <li>python38Packages.fsspec</li>
    <li>python38Packages.gcsfs</li>
    <li>python38Packages.glymur</li>
    <li>python38Packages.image-match</li>
    <li>python38Packages.imagecorruptions</li>
    <li>python38Packages.pims</li>
    <li>python38Packages.pyfftw</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.radio_beam</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.s3fs</li>
    <li>python38Packages.scikitimage</li>
    <li>python38Packages.slicedimage</li>
    <li>python38Packages.sparse</li>
    <li>python38Packages.streamz</li>
    <li>python38Packages.stumpy</li>
    <li>python38Packages.stytra</li>
    <li>python38Packages.tensorly</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.batchgenerators</li>
    <li>python39Packages.dask</li>
    <li>python39Packages.dask-gateway</li>
    <li>python39Packages.dask-image</li>
    <li>python39Packages.dask-mpi</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.distributed</li>
    <li>python39Packages.fsspec</li>
    <li>python39Packages.gcsfs</li>
    <li>python39Packages.glymur</li>
    <li>python39Packages.image-match</li>
    <li>python39Packages.imagecorruptions</li>
    <li>python39Packages.pims</li>
    <li>python39Packages.pyfftw</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.radio_beam</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.s3fs</li>
    <li>python39Packages.scikitimage</li>
    <li>python39Packages.slicedimage</li>
    <li>python39Packages.streamz</li>
  </ul>
</details>

legendofmiracles commented:
Trying to build this has now frozen my computer twice, because of what I cannot tell.

But the build log before freezing has 2 errors.

nbren12 commented:
Result of `nixpkgs-review pr 120173` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.clifford</li>
    <li>python38Packages.dask-jobqueue</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python39Packages.dask-jobqueue</li>
    <li>python39Packages.dask-xgboost</li>
  </ul>
</details>
<details>
  <summary>11 packages failed to build:</summary>
  <ul>
    <li>python38Packages.dask-glm</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.intake</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.streamz</li>
    <li>python38Packages.stumpy</li>
    <li>python39Packages.dask-glm</li>
    <li>python39Packages.intake</li>
    <li>python39Packages.spectral-cube</li>
    <li>python39Packages.streamz</li>
  </ul>
</details>
<details>
  <summary>53 packages built:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.batchgenerators</li>
    <li>python38Packages.dask</li>
    <li>python38Packages.dask-gateway</li>
    <li>python38Packages.dask-image</li>
    <li>python38Packages.dask-mpi</li>
    <li>python38Packages.datasets</li>
    <li>python38Packages.distributed</li>
    <li>python38Packages.fsspec</li>
    <li>python38Packages.gcsfs</li>
    <li>python38Packages.glymur</li>
    <li>python38Packages.image-match</li>
    <li>python38Packages.imagecorruptions</li>
    <li>python38Packages.pims</li>
    <li>python38Packages.pyfftw</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.radio_beam</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.s3fs</li>
    <li>python38Packages.scikitimage</li>
    <li>python38Packages.slicedimage</li>
    <li>python38Packages.sparse</li>
    <li>python38Packages.stytra</li>
    <li>python38Packages.sunpy</li>
    <li>python38Packages.tensorly</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.batchgenerators</li>
    <li>python39Packages.dask</li>
    <li>python39Packages.dask-gateway</li>
    <li>python39Packages.dask-image</li>
    <li>python39Packages.dask-mpi</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.distributed</li>
    <li>python39Packages.fsspec</li>
    <li>python39Packages.gcsfs</li>
    <li>python39Packages.glymur</li>
    <li>python39Packages.image-match</li>
    <li>python39Packages.imagecorruptions</li>
    <li>python39Packages.pims</li>
    <li>python39Packages.pyfftw</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.radio_beam</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.s3fs</li>
    <li>python39Packages.scikitimage</li>
    <li>python39Packages.slicedimage</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>

nbren12 committed:
python3Packages.fsspec: 0.8.3 -> 2021.04.0

other changes:
- disable test exemption for a bug that a comment claims is no longer
valid.

nbren12 committed:
python3Packages.gcsfs: init at 2021.04.0

nbren12 committed:
maintainers: add nbren12

nbren12 commented:
@legendofmiracles @SuperSandro2000 

I force-pushed a new batch of commits, splitting the maintainer addition into a new commit, and addressing the other comments.

SuperSandro2000 commented on nixos/modules/installer/tools/tools.nix:
```diff
@@ -163,7 +163,7 @@ in
         # \$ nix search wget
         # environment.systemPackages = with pkgs; [
-        #   nano vim # don't forget to add an editor to edit configuration.nix!
+        #   vim # Do not forget to add an editor to edit configuration.nix! The Nano editor is also installed by default.
         #   wget
```
I think you included this commit by accident. Please drop it.

nbren12 replied:
Weird. I re-based onto master and this diff went away.

dotlambda reviewed (changes requested):
Please add `pythonImportsCheck = [ "gcsfs" ]`.

dotlambda commented on pkgs/development/python-modules/gcsfs/default.nix:
```diff
@@ -0,0 +1,36 @@
+    owner = "dask";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-qwLi4+rNSAneQbbq2ljf58jDluVo6b/2SUwUNpFb1/Y=";
```
Hash is wrong.

nbren12 replied:
Weird. I wonder if upstream moved the tag to another commit. I've updated it.

dotlambda replied:
You should ask them. If I did, that would be reason enough for me not to allow this package into nixpkgs.

nbren12 replied:
IMO this package is professionally maintained. The hashing error was my mistake. Github shows the tag was last updated before I wrote my original commit.
